userspace/engine/rule_loader.h

```diff
@@ -33,110 +33,207 @@ class falco_engine;
 class rule_loader
 {
 public:
+	/*!
+		\brief Represents a section of text from which a certain info
+		struct has been decoded
+	*/
+	struct mark
+	{
+		std::string content;
+
+		/*!
+			\brief Wraps an error by adding info about the text section
+		*/
+		inline std::string error(std::string err)
+		{
+			err += "\n---\n";
+			err += trim(content);
+			err += "\n---";
+			return err;
+		}
+
+		/*!
+			\brief Appends another text section info to this one
+		*/
+		inline void append(mark& m)
+		{
+			content += "\n\n";
+			content += m.content;
+		}
+	};
 
 	/*!
-		\brief Erases the internal states and all the loaded rules
+		\brief Contains the info required to load rule definitions
 	*/
-	virtual void clear();
+	struct context
```

Comment:
Totally agree on this point. I like the name `context` for the _infos because 1) it respects the old naming convention and 2) I think it represents the contained info properly, and we may also expand that info in the future to improve error handling. I would propose to rename `rule_loader::context` to `rule_loader::configuration`, which I think better explains what the variables inside are for.

Reply:
yep sounds great.